### PR TITLE
doc: Bluetooth state power manager documentation

### PR DESCRIPTION
**Pull request by rakons**
Add documentation for the Bluetooth state power manager module.

JIRA: NCSDK-10556

Signed-off-by: Radoslaw Koppel <radoslaw.koppel@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/libraries/caf/ble_state_pm.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#########################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |ble_state_pm| is small module, that counts active connections and blocks power off
+if there is any active connection.
+
+Configruation
```
Comment (pdunaj) [inline doc/nrf/libraries/caf/ble_state_pm.rst:13]: Configuration

File: doc/nrf/libraries/caf/ble_state_pm.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+*************
+
+The module is enabled by selecting :kconfig:`CONFIG_CAF_BLE_STATE_PM`.
+It depends on :kconfig:`CONFIG_CAF_BLE_STATE` and :kconfig:`CONFIG_CAF_POWER_MANAGER`.
+
+There is no other configuration.
+There is no logger implemented so it cannot be configured.
```
Comment (pdunaj) [inline doc/nrf/libraries/caf/ble_state_pm.rst:20]: I would remove these two lines

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/caf/ble_state_pm.rst (hunk 1 above)
Comment (rakons) [reply doc/nrf/libraries/caf/ble_state_pm.rst:13]: ok

File: doc/nrf/libraries/caf/ble_state_pm.rst (hunk 2 above)
Comment (rakons) [reply doc/nrf/libraries/caf/ble_state_pm.rst:20]: ok

**Review by pdunaj — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]
Please do not merge before 1.7.0 release, given the cleaned up changelog file to which this PR adds an entry.

**Comment by rakons** [comment]
Waiting for release to finish - the release-notes-changelog.rst would be changed then anyway - so it makes no sense to rebase it right now.

**Comment by greg-fer** [comment]
Rebased to the latest, emptied changelog.

**Comment by greg-fer** [comment]
Removing the DNM label I put before the release.

**Comment by MarekPieta** [comment]
@rakons, can we merge this one already?

**Comment by rakons** [comment]
Yep!

**Pull request merged**